# Profitable App Profiles for the App Store and Google Play Markets
As data analysts in a company that builds Android and iOs mobile apps, we help our developers to make data-driven decisions concerning the kind of apps they make. 
Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users and hence identify the type of apps that will generate the most revenue. 

## Opening and Exploring the Data
As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play. In this project, we have 2 dataset to explore that is `AppleStore.csv` and `googleplaystore.csv`.

[Google Play Store](https://www.kaggle.com/lava18/google-play-store-apps) has approximately 10,000 Android apps.You can download the dataset [here](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv)

[The Apple App Store](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) has approximately 7,000 Android apps.You can download the dataset [here](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)
Let's start by opening the two data sets and then continue with exploring the data.

In [1]:
import csv
# The Apple Store dataset
with open("AppleStore.csv", "r", encoding="utf-8") as csv_file:
    data = csv.reader(csv_file)
    apple_apps_data = list(data)
    ios = apple_apps_data[1:]
    ios_header = apple_apps_data[0]
    
# The Google Play Store dataset
with open("googleplaystore.csv", "r", encoding="utf-8") as csv_file:
    data = csv.reader(csv_file)
    google_apps_data = list(data)
    android = google_apps_data[1:]
    android_header = google_apps_data[0]

We'll start by opening and exploring these two datasets. To make them easier to explore, we created a function named explore_data() that we can repeatedly use to print rows in a readable way.

In [2]:
def explore_data(dataset, header=False):
    for row in dataset[:3]:
        print(row)
        
    if header:
        print("\nNumber of row: " + str(len(dataset)))
        print("Number of column: " + str(len(dataset[0]))) 
        
print(ios_header)
explore_data(ios, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']
['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']
['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']

Number of row: 7197
Number of column: 16


As you can see, the Google Play dataset has 7197 apps, 16 features. Columns that maybe be useful to our analysis about the best apps in this store: `track_name`, `currency`, `price`, `rating_count_tot`, `rating_count_ver`, and `prime_genre`

In [3]:
print(android_header)
explore_data(android, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']
['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']
['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']

Number of row: 10841
Number of column: 13


As you can see, App Store dataset has 10841 apps, 13 features. Columns that maybe be useful to our analysis about the best apps in this store: `App`, `Category`, `Rating`, `Reviews`, `Installs`, `Type`, `Price` and `Genres`.

## Data cleaning
To prepare out data cleaning to analysis. We removing or correcting wrong data, removing duplicate data, and modifying the data to fit the purpose of our analysis.

### 1. Deleting incorrect data
The Google Play dataset has an error row which has the index is 10472

In [4]:
print(android_header)
print(str(android[10472])) #The error row in Google Play Store
print(str(android[0]))

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


The row 10472 corresponds to the app *Life Made WI-Fi Touchscreen Photo Frame*, and we can see that the rating is 19. This is clearly off because the maximum rating for a Google Play app is 5. The reason is the entry miss its `Category` value, which causes the column shift. As a consequence, we'll delete this row.

In [5]:
print(len(android))
del android[10472]  # don't run this more than once
print(len(android))

10841
10840


### 2. Removing duplicate entries
In the previous step, we started the data cleaning process and deleted some apps have duplicate entries from the Google Play dataset.

In [6]:
duplicate_apps = []
unique_apps = []
for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print(duplicate_apps[:5])
print(len(duplicate_apps))

['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings']
1181


We don't want to count certain apps more than once when we analyze data, so we will remove 1181 duplicate entries and keep only one entry per app. We won't remove the duplicate rows randomly, cause we have a better way.

Cause the main difference happens on the fourth position (the number of reviews) of each duplicate entries of an app, the different numbers show the data was collected at different times. Rather than removing duplicates randomly, we'll only keep the row with the highest number of reviews and remove the other entries for any given app.

In [7]:
print('Expected length: ', len(android) - len(duplicate_apps))

Expected length:  9659


We expect that our new cleaned data set , android_clean, only has unique rows/apps which is expected to be 9,659 rows. We can inspect its length to confirm this.

In [8]:
unique_apps = []
android_clean = []
for app in android:
    name = app[0]
    if name not in unique_apps:
        unique_apps.append(name)
        android_clean.append(app)
    else:
        if app[3] > android_clean[unique_apps.index(name)][3]:
            android_clean[unique_apps.index(name)] = app
print(len(android_clean))

9659


To remove the duplicate rows from the initial dataset using our lists:

1. Create two lists, one to store the newly cleaned dataset and the other to store only the app names: `android_clean`,`unique_apps`

2. Loop through the dataset, get the application name for each application and store it in a variable name `name`.

3. Inside the loop, check if the name is in the unique_apps list:
       - Append all rows to the cleaned dataset where all rows
       - Append the application name to the list containing the application name
    Otherwise: If the app reviews are more than this app's reviews in the cleaned dataset, then assign this app's data in the android dataset to this in the dataset has been cleaned.

In [9]:
explore_data(android_clean, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']
['Coloring book moana', 'FAMILY', '3.9', '974', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']
['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']

Number of row: 9659
Number of column: 13


In [10]:
reviews_max = {}
for row in android:
    name = row[0]
    n_reviews = float(row[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
android_clean = []
already_added = []
for row in android:
    name = row[0]
    n_reviews = float(row[3])
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(row)
        already_added.append(name)
print(len(android_clean))

9659


## Removing Non-English Apps
### Part One
If we explore the data long enough, we'll find that both datasets have apps with names that suggest they are not designed for an English-speaking audience.

In [11]:
print(ios[813])
print(ios[6731][1])
print('\n')
print(android_clean[4412][0])
print(android_clean[7940][0])

['445375097', '爱奇艺PPS -《欢乐颂2》电视剧热播', '224617472', 'USD', '0.0', '14844', '0', '4.0', '0.0', '6.3.3', '17+', 'Entertainment', '38', '5', '3', '1']
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


中国語 AQリスニング
لعبة تقدر تربح DZ


We're not interested in keeping these apps, so we'll remove them. One way to do this is to remove each app with a name containing a symbol that isn't commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;), and other symbols (+, *, /). 

The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the ASCII (American Standard Code for Information Interchange) system. Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not.

In [12]:
print(ord('a'))
print(ord('A'))
print(ord('爱'))
print(ord('5'))
print(ord('+'))

97
65
29233
53
43


Each character we use in a string has a corresponding number associated with it. For instance, the corresponding number for character a is 97, character A is 65, and character 爱 is 29,233. We can get the corresponding number of each character using the ord() built-in function.

### Part Two
To minimize the impact of data loss, we'll only remove an app if its name has more than three non-ASCII characters.
The function is still not perfect, and very few non-English apps might get past our filter, but this seems good enough at this point in our analysis — we shouldn't spend too much time on optimization at this point.

In [13]:
def English_app(dataset, index_name):
    Eng_apps = []
    for i in range(len(dataset)):
        count = 0
        name = dataset[i][index_name]
        for j in range(len(name)):
            character = ord(name[j])
            if character > 127:
                count += 1
        if count <= 3:
            Eng_apps.append(dataset[i])
    return Eng_apps

English_android = English_app(android_clean, 0)
English_ios = English_app(ios, 1)

explore_data(English_android, True)
print("\n")
explore_data(English_ios, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']
['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']
['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']

Number of row: 9614
Number of column: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']
['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']
['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '3

## Isolated the free apps

In [14]:
def free_apps(dataset, price_index):
    free_apps = []
    for app in dataset:
        price = app[price_index]
            
        if price == "0.0" or price == "0":
            free_apps.append(app)
    return free_apps

free_android = free_apps(English_android, 7)
free_ios = free_apps(English_ios, 4)

print(len(free_android))
print(len(free_ios))

8864
3222


We're left with 8864 Android apps and 3222 iOS apps, which should be enough for our analysis.

## Most Common Apps by Genre
### Part One
Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:
1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our datasets.

### Part Two
We'll build two functions we can use to analyze the frequency tables:
- One function to generate frequency tables that show percentages
- Another function that we can use to display the percentages in a descending order

In [15]:
def freq_table(dataset, index):
    genres = {}
    total = 0
    for app in dataset:
        genre = app[index]
        if genre in genres:
            genres[genre] += 1
            total += 1
        else:
            genres[genre] = 1

    for genre in genres:
        genres[genre] = round((genres[genre]*100/total),2)
    return genres

def display_table(dataset,index):
    display = {}
    table = freq_table(dataset,index)
    for i in range(len(table)):
        max_freq = 0
        for key in table:
            genre_count = table[key]
    #         print(genre_count)
            if genre_count > max_freq and key not in display:
                max_freq = genre_count
                genre_max = key
        display.update({genre_max:max_freq})
    return display

### Part Three
We start by examining the frequency table for the `prime_genre` column of the App Store data set.
More half of apps of App Store dataset  has genre is `Games` (58.58%), `Entertainment` is the next most common (7.94%).

We can see that the apps are designed for practical purposes (education, shopping, utilities, productivity, lifestyle) not common like entertainment (games, photo, and video, social networking, sports, music).

A large number of apps for a particular genre doesn't imply having many users, so I can't recommend an app profile for the App Store market based on this frequency table alone.

In [16]:
display_table(free_ios, 11) #Genres

{'Games': 58.58,
 'Entertainment': 7.94,
 'Photo & Video': 5.0,
 'Education': 3.69,
 'Social Networking': 3.31,
 'Shopping': 2.63,
 'Utilities': 2.53,
 'Sports': 2.16,
 'Music': 2.06,
 'Health & Fitness': 2.03,
 'Productivity': 1.75,
 'Lifestyle': 1.59,
 'News': 1.34,
 'Travel': 1.25,
 'Finance': 1.13,
 'Weather': 0.88,
 'Food & Drink': 0.81,
 'Reference': 0.56,
 'Business': 0.53,
 'Book': 0.44,
 'Navigation': 0.19,
 'Medical': 0.19,
 'Catalogs': 0.13}

Let's continue by examining the Genres and Category columns of the Google Play data set (two columns which seem to be related).

The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

In [17]:
display_table(free_android, 1) #Category

{'FAMILY': 18.98,
 'GAME': 9.76,
 'TOOLS': 8.49,
 'BUSINESS': 4.61,
 'LIFESTYLE': 3.92,
 'PRODUCTIVITY': 3.91,
 'FINANCE': 3.71,
 'MEDICAL': 3.54,
 'SPORTS': 3.41,
 'PERSONALIZATION': 3.33,
 'COMMUNICATION': 3.25,
 'HEALTH_AND_FITNESS': 3.09,
 'PHOTOGRAPHY': 2.96,
 'NEWS_AND_MAGAZINES': 2.81,
 'SOCIAL': 2.67,
 'TRAVEL_AND_LOCAL': 2.34,
 'SHOPPING': 2.25,
 'BOOKS_AND_REFERENCE': 2.15,
 'DATING': 1.87,
 'VIDEO_PLAYERS': 1.8,
 'MAPS_AND_NAVIGATION': 1.4,
 'FOOD_AND_DRINK': 1.25,
 'EDUCATION': 1.17,
 'ENTERTAINMENT': 0.96,
 'LIBRARIES_AND_DEMO': 0.94,
 'AUTO_AND_VEHICLES': 0.93,
 'HOUSE_AND_HOME': 0.83,
 'WEATHER': 0.8,
 'EVENTS': 0.71,
 'PARENTING': 0.66,
 'ART_AND_DESIGN': 0.65,
 'COMICS': 0.62,
 'BEAUTY': 0.6}

Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the Genres column:
- There're many genres in the frequency table for the `prime_genre` column of the Google Play dataset, the most commen genre is `Tools`, Tool apps are close to 8.56%
- The apps are designed for practical purposes (education, shopping, utilities, productivity, lifestyle) are really common in Google Play.

In [18]:
display_table(free_android, 9) #Genres

{'Tools': 8.56,
 'Entertainment': 6.15,
 'Education': 5.42,
 'Business': 4.65,
 'Lifestyle': 3.94,
 'Productivity': 3.94,
 'Finance': 3.75,
 'Medical': 3.58,
 'Sports': 3.51,
 'Personalization': 3.36,
 'Communication': 3.28,
 'Action': 3.14,
 'Health & Fitness': 3.12,
 'Photography': 2.98,
 'News & Magazines': 2.83,
 'Social': 2.7,
 'Travel & Local': 2.35,
 'Shopping': 2.27,
 'Books & Reference': 2.17,
 'Simulation': 2.07,
 'Dating': 1.89,
 'Arcade': 1.87,
 'Video Players & Editors': 1.79,
 'Casual': 1.78,
 'Maps & Navigation': 1.42,
 'Food & Drink': 1.26,
 'Puzzle': 1.14,
 'Racing': 1.01,
 'Libraries & Demo': 0.95,
 'Role Playing': 0.95,
 'Auto & Vehicles': 0.94,
 'Strategy': 0.93,
 'House & Home': 0.83,
 'Weather': 0.81,
 'Events': 0.72,
 'Adventure': 0.69,
 'Comics': 0.62,
 'Art & Design': 0.61,
 'Beauty': 0.61,
 'Parenting': 0.5,
 'Card': 0.46,
 'Casino': 0.43,
 'Trivia': 0.42,
 'Educational;Education': 0.4,
 'Board': 0.39,
 'Educational': 0.38,
 'Education;Education': 0.34,
 'Word

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

## Most Popular Apps by Genre on the App Store
One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the `Installs` column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the `rating_count_tot` app.

Below, we calculate the average number of user ratings per app genre on the App Store:

In [19]:
genres_ios = freq_table(free_ios, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in free_ios:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = round(total / len_genre,2)
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.35
Photo & Video : 28441.54
Games : 22788.67
Music : 57326.53
Reference : 74942.11
Health & Fitness : 23298.02
Weather : 52279.89
Utilities : 18684.46
Travel : 28243.8
Shopping : 26919.69
News : 21248.02
Navigation : 86090.33
Lifestyle : 16485.76
Entertainment : 14029.83
Food & Drink : 33333.92
Sports : 23008.9
Book : 39758.5
Finance : 31467.94
Education : 7003.98
Productivity : 21028.41
Business : 7491.12
Catalogs : 4004.0
Medical : 612.0


In [20]:
for app in free_ios:
    if app[11] == "Navigation":
        print(app[1], ":", app[5]) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [21]:
for app in free_ios:
    if app[11] == "Social Networking":
        print(app[1], ":", app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

The same pattern applies to `Social Networking` apps, **where the average number is heavily influenced by a few giants** like Facebook, Pinterest, Skype, etc. Same applies to `Music` apps, where **a few big players** like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but **Navigation, Social Networking or Music apps** might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. **We could get a better picture by removing these extremely popular apps for each genre and then rework the averages**, but we'll leave this level of detail for later.

`Reference` apps have 74,942 user ratings on average, but it's actually the **Bible and Dictionary.com** which skew up the average rating:

In [22]:
for app in free_ios:
    if app[11] == "Reference":
        print(app[1], ":", app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


However, this **niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

`Weather apps` — people generally *don't spend too much time in-app*, and the *chances of making profit from in-app adds are low*. Also, getting reliable live weather data may require us to connect our apps to *non-free APIs*.

`Food and drink` — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual *cooking and a delivery service*, which is *outside the scope of our company*.

`Finance apps` — these apps *involve banking, paying bills, money transfer, etc*. Building a finance app requires domain knowledge, and we don't want to *hire a finance expert* just to build an app.

Now let's analyze the Google Play market a bit.

## Most Popular Apps by Genre on Google Play
For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [23]:
display_table(free_android, 5) # The Intalls columns

{'1,000,000+': 15.76,
 '100,000+': 11.58,
 '10,000,000+': 10.57,
 '10,000+': 10.22,
 '1,000+': 8.41,
 '100+': 6.93,
 '5,000,000+': 6.84,
 '500,000+': 5.58,
 '50,000+': 4.78,
 '5,000+': 4.52,
 '10+': 3.55,
 '500+': 3.26,
 '50,000,000+': 2.31,
 '100,000,000+': 2.14,
 '50+': 1.92,
 '5+': 0.79,
 '1+': 0.51,
 '500,000,000+': 0.27,
 '1,000,000,000+': 0.23,
 '0+': 0.05,
 '0': 0.01}

For instance, we don't know whether an app with `100,000+` installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to find out which app genres attract the most users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on. To perform computations, however, we'll need to **convert each install number from a string to a float**. This means we need to **remove the commas and the plus characters**, or the conversion will fail and cause an error.

In [24]:
categories_android = freq_table(free_android, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in free_android:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, `Communication` apps have the most installs: **38,456,119**. This number **is heavily skewed up by a few apps that have over one billion installs** (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:`

In [25]:
for app in free_android:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we **removed all** the communication apps that have **over 100 million installs**, the average would be reduced roughly ten times:

In [27]:
under_100_m = []

for app in free_android:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

We see the same pattern for the `Video players` category, which is the runner-up with **24,727,872 installs**. The market is *dominated by apps like Youtube, Google Play Movies & TV, or MX Player*. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, **these niches seem to be dominated by a few giants who are hard to compete against**.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

**The `books and reference` genre looks fairly popular as well**, with an average number of installs of **8,767,811**. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [29]:
for app in free_android:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The `book and reference` genre includes a variety of apps: *software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc*. It seems there's still **a small number of extremely popular apps** that skew the average:

In [30]:
for app in free_android:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like **there are only a few very popular apps, so this market still shows potential**. Let's try to **get some app ideas** based on the kind **of apps** that are somewhere in the middle in terms of popularity *(between 1,000,000 and 100,000,000 downloads):*

In [32]:
for app in free_android:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

**This niche** seems to be dominated by **software for processing and reading ebooks**, as well as various collections of **libraries and dictionaries**, so it's probably not a good idea to *build similar apps since there'll be some significant competition.*

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.